pkg/allocator/allocator.go

```diff
@@ -150,7 +150,7 @@ type Allocator struct {
 	initialListDone waitChan
 
 	// idPool maintains a pool of available ids for allocation.
-	idPool *idpool.IDPool
+	idPool idpool.IDPool
```

Comment:
I don't think this changes anything WRT go's GC because the variable is only initialized once when the agent starts.

Reply:
True; in this specific case, the impact in minimal. The general rule I try to follow is to only make a variable a pointer if it must be one. I noticed while looking at this code, it didn't need to be a pointer, and changing it so was quite trivial. I think moving forward we should also consider reducing our usage of pointers. Over time, we may find that we improve the performance of Cilium because it'll reduce the load on the GC.